### PR TITLE
BUG: fix problem with modifing pyf lines containing ';' in f2py

### PR DESCRIPTION
**Pull request by bobeldering**
This fix includes the patch as suggested by @WarrenWeckesser in #10676 and a basic regression test.

### CODE REVIEW

File: numpy/f2py/tests/test_semicolon_split.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+python module {module}
+    usercode '''
+void foo(int* x) {{
+}}
+'''
+    interface
+        subroutine foo(x)
+	    intent(c) foo
+            integer intent(out) :: x
+	    callprotoargument int*
```
Comment (charris) [inline numpy/f2py/tests/test_semicolon_split.py]: Indentation?

Comment (charris) [reply numpy/f2py/tests/test_semicolon_split.py]: Funny, that looks good in the code. Perhaps there are some hard tabs?

Comment (charris) [reply numpy/f2py/tests/test_semicolon_split.py]: Yep, you have a bunch of hard tabs scattered in the file. A `retab` should clean this up in vim.

Comment (charris) [reply numpy/f2py/tests/test_semicolon_split.py]: OK, there are other files with this problem. I'll put this in and make a fixup PR.

**Comment by charris** [comment]
Thanks @bobeldering .

**Comment by bobeldering** [comment]
Mashing the tab button is such an automatic reflex, unfortunately not such a good idea in text ... sorry for the mess.

**Comment by charris** [comment]
No problem. You should be able to control hard vs soft tabs with an editor setting.

**Comment by charris** [comment]
@bobeldering Just a note that your fixes to f2py won't show up in a release until about the time SciPy 1.2 comes out. That is to work around the SciPy bug because once the fix is released NumPy will be unable to compile SciPy < 1.0.1.

**Pull request merged**